react/training-project/src/app/screens/Login/index.js

```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+import actionCreators from '~redux/Auth/actions';
+
+import LoginForm from './layout';
+
+/* eslint-disable no-func-assign */
+class LoginPage extends Component {
+  handleSubmit = values => {
+    this.props.logIn(values);
+  };
+
+  render() {
+    return !this.props.token ? (
+      <LoginForm
+        processing={this.props.processing}
+        loginFail={this.props.loginFail}
+        onSubmit={this.handleSubmit}
+      />
+    ) : (
+      <Redirect to="/app/game" />
```

Comment:
Use constants for paths. Create a constants.js file in the Routes folder.

Reply:
already solved in PR19